cg/src/domains/subjects/subjects.service.js

```diff
@@ -29,27 +28,23 @@ class SubjectsService {
   }
 
   async registerConsentToProcessData(subjectId, personalData, processorIds = []) {
+    let processorIdsWithAddresses;
     await this.db.transaction(async trx => {
```

Comment:
We could refactor this out to a separate method that returns consented processors, but since we're likely to change this approach, then we should do that later.

Reply:
I also think we should refactor this service, because it's hard to maintain this piece of the code. Every time I need to work on it, I have to go about understanding it all again.
In the task to create the update consent endpoint, I will try to separate it in 2 methods